android/app/src/main/java/fct/inesctec/stayaway/tracing/internal/networking/ConfigWorker.java

```diff
@@ -120,6 +121,21 @@ public void loadConfig() throws IOException, ResponseError, SignatureException {
                 config.getParameters().getLowerThreshold(), config.getParameters().getHigherThreshold(),
                 config.getParameters().getFactorLow(), config.getParameters().getFactorHigh(),
                 config.getParameters().getTriggerThreshold());
+
+        // Check InfoBox
+        SecureStorage secureStorage = SecureStorage.getInstance(context);
+        InfoBoxModel info = config.getInfoBox(context.getString(R.string.language_key));
+        if (info != null) {
+            if (info.getId() == null || !info.getId().equals(secureStorage.getInfoboxId())) {
```

Comment:
If there's no else condition can't we merge the two if conditions into one?

if( info != null &&  
  (info.getId() == null || !info.getId().equals(secureStorage.getInfoboxId())

It will always check the info != null before going forward with the next condition

Reply:
Right, we will change before merging.